### ISSUE TITLE
Kafka Unhealthy

### ISSUE THREAD

**Issue by Dec-** [strimzi/strimzi-kafka-operator#518]
Hi,

on big load on ur kafka cluster ur readiness and liveness probe failed with error:

`Liveness probe failed: Exception in thread "main" java.lang.RuntimeException: Request METADATA failed on brokers List(localhost:9092 (id: -1 rack: null)) at kafka.admin.AdminClient.sendAnyNode(AdminClient.scala:107) at kafka.admin.AdminClient.findAllBrokers(AdminClient.scala:165) at kafka.admin.AdminClient.awaitBrokers(AdminClient.scala:157) at kafka.admin.BrokerApiVersionsCommand$.execute(BrokerApiVersionsCommand.scala:42) at kafka.admin.BrokerApiVersionsCommand$.main(BrokerApiVersionsCommand.scala:36) at kafka.admin.BrokerApiVersionsCommand.main(BrokerApiVersionsCommand.scala)`

`Readiness probe failed: [2018-06-22 20:52:11,792] WARN Bootstrap broker localhost:9092 (id: -1 rack: null) disconnected (org.apache.kafka.clients.NetworkClient) Exception in thread "main" java.lang.RuntimeException: Request METADATA failed on brokers List(localhost:9092 (id: -1 rack: null)) at kafka.admin.AdminClient.sendAnyNode(AdminClient.scala:107) at kafka.admin.AdminClient.findAllBrokers(AdminClient.scala:165) at kafka.admin.AdminClient.awaitBrokers(AdminClient.scala:157) at kafka.admin.BrokerApiVersionsCommand$.execute(BrokerApiVersionsCommand.scala:42) at kafka.admin.BrokerApiVersionsCommand$.main(BrokerApiVersionsCommand.scala:36) at kafka.admin.BrokerApiVersionsCommand.main(BrokerApiVersionsCommand.scala)`

Is anyone else running into the same problem?

**Comment by scholzj** [comment]
Thanks for raising this. I think we might need to fine tune the configuration. But we should probably also think about some more lightweight healthcheck.

When you experienced these problems ... did you experienced also some similar problems in your clients which were connected and creating the load?

**Comment by Dec-** [comment]
It looks like if one of the broker fails probe test all of my broker become unavailable? Really strange behavior.

**Comment by Dec-** [comment]
So what I did next is i removed all probes from kafka-brokers and everything works fine for like 30min, after that my kafka-producer can't connect to my strimzi cluster (timeout).

Can some1 help?

**Comment by scholzj** [comment]
How did you removed the probes? The Cluster Operator should normally recreate them. If the client cannot connect, you might want to check:
* The Pod readiness
* The service endpoints

Are the Pods ready and do the services have the brokers as endpoints?

**Comment by Dec-** [comment]
We put down the cluster operator just for test. Then removed Liveness and Readiness probe. After that pods are available. At that point we manually went to every pod and executed healthcheck, and everything was OK on every pod. Then we started creating topics (20k of them in 20 minutes). Topic are created and we can see them in Zookeeper. After some period of time pods are not restarted but unavailable. There is no errors in broker logs.

For broker Java options we use -Xmx 8g i -Xms 8g . We have 5 brokers. That are running on 3 application node that has 32 cores and 300gb of memory.
For persistent volumes we use disks on NetApp storage provided with iSCSI protocol. The peek is around 3.5k iops, and after some time it drops on 500 iops, and then pods/kafka-brokers become unavailable.
